drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,371 @@
+*    patent holders to use this software.
+*  - Use of the software either in source or binary form, must be run
+*    on or directly connected to an Analog Devices Inc. component.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+* MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "no_os_util.h"
+#include "no_os_delay.h"
+#include "no_os_alloc.h"
+#include "no_os_i2c.h"
+
+#include "ltc3350.h"
+
+/******************************************************************************/
+/************************ Variable Declarations ******************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/***************************************************************************//**
+
+/***************************************************************************//**
+ * @brief Initializes the communication peripheral and checks if the LTC3350
+ *        part is present.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int ltc3350_init(struct ltc3350_dev **device,
+		 struct ltc3350_init_param *init_param)
+{
+	struct ltc3350_dev *dev;
+	int ret = 0;
+	dev = (struct ltc3350_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_i2c_init(&dev->i2c_desc, &init_param->i2c_init);
+
```

Comment:
no extra line needed here. the convention is that after each error code return we do a check like this:
```
if (ret)
  return ret;
```

There are multiple drivers already on the master branch that use this convetion. Check them out. Here are some examples:
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/power/ltc3337/ltc3337.c
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/adc-dac/ad74416h/ad74416h.c